src/api/index.test.js

```diff
@@ -1,56 +1,234 @@
+      username: 'Celeste',
+      email,
+      contacts: [{}],
+      role: 'basic',
+    };
+    const response = {
+      success: true,
+      message: 'Logged in successfully !',
+      token: 'TestToken121212',
+      user,
+    };
+    // when
+    mockFMClient.onPost('/login', { email, password }).reply(201, response);
+    const actual = await appApiClient.loginUser(email, password);
+    // then
+    expect(actual.data).toEqual(response);
+  });
+
+  it('should successfully send user data on POST to /signup endpoint', async () => {
+    const email = 'test@test.com';
+    const password = '12345678';
+    const username = 'celeste';
+    const token = 'TestToken121212';
+    // given
+    const user = {
+      username,
+      email,
+      contacts: [{}],
+      role: 'basic',
+    };
+    const response = {
+      success: true,
+      token,
+      user,
+    };
+    // when
+    mockFMClient
+      .onPost('/signup', { email, password, username })
+      .reply(201, response);
+    const actual = await appApiClient.signupUser(email, password, username);
+    // then
+    expect(actual.data).toEqual(response);
+  });
+  it('should successfully delete user on DELETE request to /deleteUser endpoint', async () => {
+    const username = 'celeste';
+    const email = 'test@test.com';
+    // given
+    const user = {
+      username,
+      email,
+      contacts: [{}],
+      role: 'basic',
+    };
+    const response = {
+      message: 'User was deleted',
+      user,
+    };
+    // when
+    mockFMClient
+      .onDelete('/deleteUser', { params: { username } })
+      .reply(200, response);
+    const actual = await appApiClient.deleteUser(username);
+    // then
+    expect(actual.data).toEqual(response);
+  });
+
+  it('should successfully change password of user on POST request to /changePassword endpoint', async () => {
+    const email = 'test@test.com';
+    const password = '12345678';
+    const oldPassword = '87654321';
+
+    // given
+    const response = {
+      message: 'You updated the password',
+    };
+    // when
+    mockFMClient
+      .onPost('/changePassword', { email, oldPassword, password })
+      .reply(200, response);
+    const actual = await appApiClient.changePassword(
+      email,
+      oldPassword,
+      password
+    );
+    // then
+    expect(actual.data).toEqual(response);
+  });
+
+  it('should successully get sos contacts with username and token', async () => {
+    // given
+    const username = 'celeste';
+    const token = 'faketoken123456';
+    const response = [
+      {
+        _id: '2f213dsafdsfasdfdas34e',
+        name: 'Soyoon',
+        phone: '012341235215',
+        message: 'help',
+      },
+    ];
+    mockFMClient.onGet(`/users/${username}/contacts`).reply(200, response, {
+      headers: { 'auth-token': token },
+    });
+    // when
+    const contacts = await appApiClient.getSosContacts(username, token);
+    // then
+    expect(contacts.data).toEqual(response);
   });
```

Comment:
looks good ! do we need to test fail cases on the frontend too ?

Reply:
yes, i think we can send errors and show how FE response to it